### PR TITLE
feat: add plugin setting support

### PR DESCRIPTION
**Pull request by guqing**
#### What type of PR is this?
/kind feature
/area core
/milestone 2.0

#### What this PR does / why we need it:
新增插件设置，允许通过 Setting 自定义模型生成表单并收集其值为 ConfigMap
插件可以通过注入 SettingFetcher 来获取值并使用它,例如：
```java
Sns sns = settingFetcher.getGroupForObject("sns", Sns.class);
```
#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:
/cc @halo-dev/sig-halo 
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by ruibaby** [comment]
admin 已经在 https://github.com/halo-dev/halo-admin/commit/dc1da8fc91c7a356a135ca5a1a7b292fc7073938 提交中适配该特性。

File: src/main/java/run/halo/app/core/extension/Plugin.java [hunk 1]
```diff
@@ -64,6 +64,10 @@ public static class PluginSpec {
         private Boolean enabled = false;
 
         private List<String> extensionLocations;
+
+        private String settingName;
+
+        private String configmapName;
```
Comment (JohnNiang) [inline src/main/java/run/halo/app/core/extension/Plugin.java:70]: configmapName or configMapName?

Comment (guqing) [reply src/main/java/run/halo/app/core/extension/Plugin.java:70]: > configmapName or configMapName?

考虑到宽松绑定，驼峰可以写成短横线对于 configmapName可以写成configmap-name而configMapName只能写成config-map-name就这么写了，configMapName也行

**Review by JohnNiang — APPROVED** [review]
/approve

**Review by JohnNiang — COMMENTED** [review]
/lgtm

**Pull request merged**